### PR TITLE
chests: fix sort_inventory function

### LINKED ISSUE #381: sort_inventory destroys item metadata!
[`get_metadata()`](https://github.com/minetest-mods/technic/blob/master/technic_chests/register.lua#L137) is deprecated, because an item can have more then one metadata!
This issue is painful, because my [inventorybags mod](https://forum.minetest.net/viewtopic.php?f=11&t=17847) is using the metadata as item storage.

### PR DESCRIPTION
**Pull request by DS-Minetest**
Fixes #381.

### CODE REVIEW

File: technic_chests/register.lua [hunk 1]
```diff
@@ -137,32 +137,15 @@ local function sort_inventory(inv)
...
 			local k = string.format("%s %05d %s", n, w, m)
 			if not typecnt[k] then
-				typecnt[k] = {
-					name = n,
-					wear = w,
-					metadata = m,
-					stack_max = st:get_stack_max(),
-					count = 0,
-				}
+				typecnt[k] = st:to_string()
```
Comment (numberZero) [inline technic_chests/register.lua]: Is this conversion necessary?

File: technic_chests/register.lua [hunk 2]
```diff
@@ -137,21 +137,25 @@ local function sort_inventory(inv)
...
+	local empty_list = {}
+	for i = 1, #inlist do
+		empty_list[i] = ItemStack(nil)
 	end
-	if #outlist > #inlist then return end
-	while #outlist < #inlist do
-		table.insert(outlist, ItemStack(nil))
+	inv:set_list("main", empty_list)
+	for i = 1, #typekeys do
+		for k = 1, #typecnt[typekeys[i]] do
```
Comment (numberZero) [inline technic_chests/register.lua]: er... `ipairs`?

File: technic_chests/register.lua [hunk 3]
```diff
@@ -137,21 +137,25 @@ local function sort_inventory(inv)
...
+			else
+				table.insert(typecnt[k], st)
 			end
 		end
 	end
 	table.sort(typekeys)
-	local outlist = {}
-	for _, k in ipairs(typekeys) do
-		table.insert(outlist, ItemStack(typecnt[k]))
+	if #typekeys > #inlist then return end
```
Comment (numberZero) [inline technic_chests/register.lua]: impossible

File: technic_chests/register.lua [hunk 4]
```diff
@@ -137,21 +137,25 @@ local function sort_inventory(inv)
...
+	local empty_list = {}
+	for i = 1, #inlist do
+		empty_list[i] = ItemStack(nil)
 	end
-	if #outlist > #inlist then return end
-	while #outlist < #inlist do
-		table.insert(outlist, ItemStack(nil))
+	inv:set_list("main", empty_list)
```
Comment (numberZero) [inline technic_chests/register.lua]: `{}` should work as well

File: technic_chests/register.lua (hunk 2 above)
Comment (DS-Minetest) [reply technic_chests/register.lua]: Is `ipairs` faster?

File: technic_chests/register.lua (hunk 3 above)
Comment (DS-Minetest) [reply technic_chests/register.lua]: The check already was there before. I'll remove it.

File: technic_chests/register.lua (hunk 2 above)
Comment (numberZero) [reply technic_chests/register.lua]: it’s not that ugly, at least

Comment (DS-Minetest) [reply technic_chests/register.lua]: Imho this is not ugly.

**Comment by numberZero** [comment]
Wow. It’s that short now.
Really, if it works as expected, it should be merged ASAP IMO.
:+1:

**Comment by DS-Minetest** [comment]
Would this please be merged?

**Comment by numberZero** [comment]
@Ekdohibs ?

**Comment by Ekdohibs** [comment]
https://github.com/minetest-mods/technic/commit/930c51eef954d436e559b8b67b3c682d6c0865fb

**Pull request closed without merging**